### PR TITLE
Add plot recipe for initial condition

### PR DESCRIPTION
**Pull request by LasNikas**
with `hydrostatic_water_column_2d.jl` example

`plot(sol)`:
image

`plot(tank.fluid, tank.boundary)`:
image

### CODE REVIEW

File: src/visualization/recipes_plots.jl [hunk 1]
```diff
@@ -67,3 +67,53 @@ RecipesBase.@recipe function f(v_ode, u_ode, semi::Semidiscretization;
...
+
+        label --> label
+
+        colorbar_title --> colorbar_title
+
+        zcolor --> zcolor
+
+        # Return data for plotting
+        x, y
+    end
```
Comment (efaulhaber) [inline src/visualization/recipes_plots.jl:118]: All of this is duplicate code. Can you please make it a function?

**Review by efaulhaber — CHANGES_REQUESTED** [review]

File: src/visualization/recipes_plots.jl [hunk 2]
```diff
@@ -29,10 +29,38 @@ RecipesBase.@recipe function f(v_ode, u_ode, semi::Semidiscretization;
...
+
+        return (; x, y, x_min, x_max, y_min, y_max, particle_spacing,
+                label="initial condition " * "$idx")
+    end
+
+    return (first(initial_conditions), ics...)
+end
+
+RecipesBase.@recipe function f(::Union{InitialCondition, Semidiscretization},
+                               data...; zcolor=nothing, size=(600, 400), colorbar_title="")
```
[the remark this answers is not in the document]
Comment (LasNikas) [reply src/visualization/recipes_plots.jl:59]: yes

[the remark this answers is not in the document]
Comment (LasNikas) [reply src/visualization/recipes_plots.jl:59]: `plot(sol; size=(1200,300), zcolor=tank.fluid.pressure, colorbar_title="test")`
image

[the remark this answers is not in the document]
Comment (LasNikas) [reply src/visualization/recipes_plots.jl:59]: `plot(tank.fluid; size=(1200,300), zcolor=tank.fluid.pressure, colorbar_title="test")`
image

[the remark this answers is not in the document]
Comment (LasNikas) [reply src/visualization/recipes_plots.jl:59]: even for other plot-kwargs
`plot(tank.fluid; size=(1200,300), zcolor=tank.fluid.pressure, colorbar_title="test",xaxis=false, yaxis=false)`
image

**Review by efaulhaber — APPROVED** [review]

**Pull request merged**